butane_core/src/migrations/adb.rs

```diff
@@ -318,19 +329,67 @@ impl From<TypeIdentifier> for DeferredSqlType {
     }
 }
 
+/// Abstract representation of a database column reference constraint.
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+pub enum ARef {
+    /// A reference to a literal table column.
+    Literal(ARefLiteral),
+    /// A reference that has not been resolved yet.
+    Deferred(DeferredSqlType),
+}
+
+/// Abstract representation of a database column reference constraint to a literal table and column.
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+pub struct ARefLiteral {
+    /// Table name.
+    table_name: String,
+    /// Column name.
+    column_name: String,
+}
+
+impl ARefLiteral {
+    /// Create new literal reference to a table and column.
+    pub fn new(table_name: impl Into<String>, column_name: impl Into<String>) -> Self {
+        ARefLiteral {
+            table_name: table_name.into(),
+            column_name: column_name.into(),
+        }
+    }
+    /// Get table name.
+    pub fn table_name(&self) -> &str {
+        &self.table_name
+    }
+    /// Get column name.
+    pub fn column_name(&self) -> &str {
+        &self.column_name
+    }
+}
+
 /// Abstract representation of a database column schema.
 #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
 pub struct AColumn {
+    /// Column name.
     name: String,
+    /// Type of the column.
     sqltype: DeferredSqlType,
+    /// Whether the column is nullable.
     nullable: bool,
+    /// Whether the column is a primary key.
     pk: bool,
+    /// Whether the column is an auto-increment field.
     auto: bool,
+    /// Whether the column needs a unique constraint.
     #[serde(default)]
     unique: bool,
+    /// Default value for the column.
     default: Option<SqlVal>,
+    /// Whether this column refers to another column.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    reference: Option<ARef>,
 }
 impl AColumn {
+    /// Create new column.
+    #[allow(clippy::too_many_arguments)]
```

Comment:
This should probably be a builder, which would remove the need for the clippy allow and make invocations more readable. Definitely doesn't need to be this PR though, I can try to address it afterwards.